### PR TITLE
[Misc][TPU] Support TPU in initialize_ray_cluster

### PR DESCRIPTION
**Pull request by WoosukKwon**
This PR adds support for TPUs in initializing Ray cluster.

Will also add HPU in the PR #6143

### CODE REVIEW

**Comment by WoosukKwon** [comment]
This PR was part of https://github.com/vllm-project/vllm/pull/5871, but separated out to get a quick review.

File: vllm/executor/ray_utils.py [hunk 1]
```diff
@@ -101,24 +102,27 @@ def initialize_ray_cluster(
         # Verify that we can use the placement group.
         gpu_bundles = 0
```
Comment (rkooo567) [inline vllm/executor/ray_utils.py:103]: probably need to change this name!

File: vllm/executor/ray_utils.py [hunk 2]
```diff
@@ -101,24 +102,27 @@ def initialize_ray_cluster(
...
+                    "Placement group bundle cannot have more than 1 "
+                    f"{device_str}.")
             if bundle_gpus:
                 gpu_bundles += 1
         if parallel_config.world_size > gpu_bundles:
             raise ValueError(
-                "The number of required GPUs exceeds the total number of "
-                "available GPUs in the placement group.")
+                f"The number of required {device_str}s exceeds the total "
+                f"number of available {device_str}s in the placement group.")
```
Comment (rkooo567) [inline vllm/executor/ray_utils.py:115]: nit: but the error message will be nicer if we also include `parallel_config.world_size` and `gpu_bundles`. 

```
The number of required {device_str}s exceeds the total number of available {device_str}s in the placement groups. Required number of devices: {parallel_config.world_size}. Total number of devices: {gpu_bundles}.
```

File: vllm/executor/ray_utils.py [hunk 3]
```diff
@@ -101,24 +102,27 @@ def initialize_ray_cluster(
...
     else:
-        num_gpus_in_cluster = ray.cluster_resources().get("GPU", 0)
+        num_gpus_in_cluster = ray.cluster_resources().get(device_str, 0)
         if parallel_config.world_size > num_gpus_in_cluster:
```
Comment (rkooo567) [inline vllm/executor/ray_utils.py:118]: var name!

File: vllm/executor/ray_utils.py [hunk 4]
```diff
@@ -101,24 +102,27 @@ def initialize_ray_cluster(
...
-            bundle_gpus = bundle.get("GPU", 0)
+            bundle_gpus = bundle.get(device_str, 0)
             if bundle_gpus > 1:
                 raise ValueError(
-                    "Placement group bundle cannot have more than 1 GPU.")
+                    "Placement group bundle cannot have more than 1 "
+                    f"{device_str}.")
             if bundle_gpus:
                 gpu_bundles += 1
         if parallel_config.world_size > gpu_bundles:
```
Comment (rkooo567) [inline vllm/executor/ray_utils.py:112]: var name!

**Review by rkooo567 — APPROVED** [review]
LGTM. small nits

File: vllm/executor/ray_utils.py (hunk 2 above)
Comment (WoosukKwon) [reply vllm/executor/ray_utils.py]: Sounds good! Added.

File: vllm/executor/ray_utils.py (hunk 1 above)
Comment (WoosukKwon) [reply vllm/executor/ray_utils.py]: Nice catch! Fixed the variable names from gpu* -> device*

**Pull request merged**